src/map/script.c

```diff
@@ -2680,6 +2680,13 @@ struct script_data *get_val(struct script_state* st, struct script_data* data) {
 	prefix = name[0];
 	postfix = name[strlen(name) - 1];
 
+	if (strlen(name) > 32) {
+		ShowError("script_get_val: variable name too long. '%s'\n", name);
```

Comment:
already tested it can save in SQL with 32 characters of $zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz$
but maybe because it is `VARCHAR(32) BINARY` and not `VARCHAR(32)`